internal/eval/eval.go

```diff
@@ -2733,6 +2733,43 @@ func evalHSET(args []string, store *dstore.Store) []byte {
 	return clientio.Encode(numKeys, false)
 }
 
+func evalHINCRBY(args []string, store *dstore.Store) []byte {
+	if len(args) < 3 {
+		return diceerrors.NewErrArity("HINCRBY")
+	}
+
+	key := args[0]
+	obj := store.Get(key)
+	var hashmap HashMap
+
+	if obj != nil {
+		if err := object.AssertTypeAndEncoding(obj.TypeEncoding, object.ObjTypeHashMap, object.ObjEncodingHashMap); err != nil {
+			return diceerrors.NewErrWithMessage(diceerrors.WrongTypeErr)
+		}
+		hashmap = obj.Value.(HashMap)
+	}
+
+	if hashmap == nil {
+		hashmap = make(HashMap)
+	}
+
+	field := args[1]
+	increment, err := strconv.ParseInt(args[2], 10, 64)
+	if err != nil {
+		return diceerrors.NewErrWithFormattedMessage(diceerrors.IntOrOutOfRangeErr)
+	}
```

Comment:
Hi, upon check the redis behavior for handling this command, I think we can order the validation to ensure we are consistent with the output when there are multiple issues with the command. The order in redis is:
1. Check arity
2. Check value is integer
3. Get or Create object and validate type
4. Check field is integer
5. perform operation
I have following suggestion, if you would consider it:
```suggestion

	field := args[1]
	increment, err := strconv.ParseInt(args[2], 10, 64)
	if err != nil {
		return diceerrors.NewErrWithFormattedMessage(diceerrors.IntOrOutOfRangeErr)
	}
	key := args[0]
	obj := store.Get(key)
	var hashmap HashMap

	if obj != nil {
		if err := object.AssertTypeAndEncoding(obj.TypeEncoding, object.ObjTypeHashMap, object.ObjEncodingHashMap); err != nil {
			return diceerrors.NewErrWithMessage(diceerrors.WrongTypeErr)
		}
		hashmap = obj.Value.(HashMap)
	}

	if hashmap == nil {
		hashmap = make(HashMap)
	}


```

Reply:
Why does the ordering of the validations matter here? All the validations are independent of each other. So. the ordering shouldn't effect the operation being performed.

Reply:
Hi, the ordering of the validations performed  will decide what error message you will be returning when multiple validations fail. For instance, lets consider the following run in redis:
![image](https://github.com/user-attachments/assets/7e79e6c2-4c07-40a3-afb9-8280aefd9b61)

when I run the command `hincrby a` and `hincrby a v`. It returns Arity error. 

If I run the command `hincrby a v a`, it returns integer error. In your case, since the object does not exist, it will create and then check and then return integer error.

However, If I create an object which is of different type using cmd `set a b` and then run again, as per your logic, it will return type error, but as you see in the output it returns integer error.

For reference please check code for redis [here](https://github.com/redis/redis/blob/5f28bd96dbe67e5d924d5bbab3ada5618ec52490/src/t_hash.c#L2187).